### ISSUE TITLE
[M2-P12] Wall Loss builders and simulation notebooks - API migration

### ISSUE THREAD

**Issue by Gorkowski** [uncscode/particula#926]
Dependencies:

#925 [M2-P11] ──► #THIS [M2-P12]

**Parent Issue:** #914

**Dependencies:**
- M2-P11: #925

---

## Description

Migrate the wall loss builder and simulation notebooks to use current API patterns. These notebooks cover builder/factory patterns and forward simulation for chamber wall loss.

**Part of Maintenance Plan M2:** Jupyter Notebook API Migration (Parent: #914)

## Context

This phase covers 2 notebooks in the chamber wall loss directory:
1. `wall_loss_builders_factory.ipynb` - Builder and factory pattern demonstrations
2. `Chamber_Forward_Simulation.ipynb` - Full forward simulation example

These notebooks demonstrate the modern builder/factory API and complete simulation workflows.

**Value:**
- Demonstrates current builder/factory patterns
- Provides complete forward simulation example
- Completes wall loss notebook migration (P11-P12)

## Scope

**Estimated Lines of Code:** ~40 lines (API updates across 2 notebooks)
**Complexity:** Small (focused API updates)

**Files to Modify:**
- `docs/Examples/Chamber_Wall_Loss/Notebooks/wall_loss_builders_factory.ipynb`
- `docs/Examples/Chamber_Wall_Loss/Notebooks/Chamber_Forward_Simulation.ipynb`

## Pre-Phase Validation (REQUIRED)

Before making any changes, run each notebook to understand current state:

```python
# Test both wall loss builder/simulation notebooks
run_notebook({notebookPath: "docs/Examples/Chamber_Wall_Loss/Notebooks/wall_loss_builders_factory.ipynb", timeout: 300})
run_notebook({notebookPath: "docs/Examples/Chamber_Wall_Loss/Notebooks/Chamber_Forward_Simulation.ipynb", timeout: 300})
```

If a notebook already passes, document as "verified - no changes needed" but still include in PR for completeness.

### Core Implementation
- [ ] Update all deprecated API patterns to current builders/factories
- [ ] Ensure all imports use current module structure
- [ ] Update descriptions and learning objectives for clarity

### Per-Notebook Validation (Co-located with implementation)
- [ ] `wall_loss_builders_factory.ipynb` - `run_notebook` passes
- [ ] `Chamber_Forward_Simulation.ipynb` - `run_notebook` passes
- [ ] No deprecated imports in either notebook
- [ ] Both notebooks use current builder/factory patterns

### Documentation
- [ ] Each notebook has clear learning objectives
- [ ] Output cells cleared before commit

### API Migration Patterns to Apply

These notebooks should showcase the current builder/factory patterns:

```python
# Builder Pattern
strategy = (
    par.dynamics.ChargedWallLossBuilder()
    .set_wall_eddy_diffusivity(0.001)
    .set_chamber_geometry("spherical")
    .set_chamber_radius(0.5)
    .set_wall_potential(0.05)
    .set_wall_electric_field(0.0)
    .build()
)

# Factory Pattern
strategy = par.dynamics.WallLossFactory().get_strategy(
    strategy_type="charged",
    parameters={...}
)

# WallLoss Runnable
wall_loss = par.dynamics.WallLoss(wall_loss_strategy=strategy)
aerosol = wall_loss.execute(aerosol, time_step=1.0, sub_steps=2)
```

### Notebooks Overview

1. **wall_loss_builders_factory.ipynb**
   - Builder pattern demonstrations
   - Factory pattern demonstrations
   - Strategy configuration options

2. **Chamber_Forward_Simulation.ipynb**
   - Complete chamber simulation setup
   - Time evolution with wall loss
   - Results visualization

### Validation Commands
```python
# Test wall loss builder/simulation notebooks
run_notebook({notebookPath: "docs/Examples/Chamber_Wall_Loss/Notebooks/wall_loss_builders_factory.ipynb", timeout: 300})
run_notebook({notebookPath: "docs/Examples/Chamber_Wall_Loss/Notebooks/Chamber_Forward_Simulation.ipynb", timeout: 300})
```

### CI Behavior
- `.github/workflows/notebooks.yml` validates on PR
- 5-minute timeout per notebook
- Both notebooks should complete within timeout

## References

**Maintenance Plan:**
- `adw-docs/dev-plans/maintenance/M2-notebook-api-migration.md`

**CI Infrastructure:**
- `.github/workflows/notebooks.yml` - Notebook validation workflow

**AGENTS.md Wall Loss Reference:**
- See "Wall loss strategies" section in AGENTS.md for current API patterns

**Source Notebooks:**
- `docs/Examples/Chamber_Wall_Loss/Notebooks/wall_loss_builders_factory.ipynb`
- `docs/Examples/Chamber_Wall_Loss/Notebooks/Chamber_Forward_Simulation.ipynb`

**Dependencies:**
- #925 - M2-P11 (wall loss tutorial and strategies)

**Parent Issue:**
- #914 - M2: Jupyter Notebook API Migration
